src/test/java/com/example/spring/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,156 @@
+
+        Book book = new Book();
+        book.setId(1L);
+
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setId(1L);
+
+        final RequestCartItemDto requestCartItemDto = new RequestCartItemDto(1L, 1);
+
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItem.setQuantity(1);
+        when(bookRepository.findById(requestCartItemDto.bookId())).thenReturn(Optional.of(book));
+        when(shoppingCartRepository.findByUserEmail(user.getEmail())).thenReturn(shoppingCart);
+        when(cartItemRepository.existsByBook(book)).thenReturn(false);
+        when(cartItemMapper.toModel(requestCartItemDto)).thenReturn(cartItem);
+        when(cartItemRepository.save(cartItem)).thenReturn(cartItem);
+        when(cartItemMapper.toDto(cartItem)).thenReturn(
+                new CartItemDto(1L, book.getId(), book.getTitle(), 1));
+
+        // When
+        CartItemDto result = shoppingCartService.addCartItem(user, requestCartItemDto);
+
+        // Then
+        assertNotNull(result);
+        assertEquals(1L, result.bookId().longValue());
+        assertEquals(1, result.quantity());
+        verify(cartItemRepository, times(1)).save(cartItem);
+    }
+
+    @Test
+    @DisplayName("Should throw EntityNotFoundException when not existing book is provided")
+    public void addCartItem_NotExistingBook_ShouldEntityNotFoundException() {
+        // Given
+        User user = new User();
+        user.setEmail("user@example.com");
+
+        RequestCartItemDto requestCartItemDto = new RequestCartItemDto(1L, 1);
+
+        when(bookRepository.findById(requestCartItemDto.bookId())).thenReturn(Optional.empty());
+
+        // When & Then
+        assertThrows(EntityNotFoundException.class,
+                () -> shoppingCartService.addCartItem(user, requestCartItemDto));
+    }
+
+    @Test
+    @DisplayName("Should throw DataProcessingException when existing cart item is provided")
+    public void addCartItem_NotExistingBook_ShouldThrowDataProcessingException() {
+        // Given
+        User user = new User();
+        user.setEmail("user@example.com");
+
+        Book book = new Book();
+        book.setId(1L);
+
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setId(1L);
+
+        RequestCartItemDto requestCartItemDto = new RequestCartItemDto(1L, 1);
+
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+
+        when(bookRepository.findById(requestCartItemDto.bookId())).thenReturn(Optional.of(book));
+        when(shoppingCartRepository.findByUserEmail(user.getEmail())).thenReturn(shoppingCart);
+        when(cartItemRepository.existsByBook(book)).thenReturn(true);
+        when(cartItemRepository.findByBook(book)).thenReturn(List.of(cartItem));
+
+        // When & Then
+        assertThrows(DataProcessingException.class,
```

Comment:
Don't test system exceptions! 
I guess you don't have any logic for handling this exception

Reply:
Please, check this review comment about system exceptions